tests/test_interpolation.py

```diff
@@ -477,3 +714,455 @@ def test_optional_after_interpolation() -> None:
     # Ensure that we can set an optional field to `None` even when it currently
     # points to a non-optional field.
     cfg.opt_num = None
+
+
+def test_empty_stack() -> None:
+    """
+    Check that an empty stack during ANTLR parsing raises a `GrammarParseError`.
+    """
+    with pytest.raises(GrammarParseError):
+        grammar_parser.parse("ab}", lexer_mode="VALUE_MODE")
+
+
+def _maybe_create(definition: str) -> Any:
+    """
+    Helper function to create config objects for lists and dictionaries.
+    """
+    if isinstance(definition, (list, dict)):
+        return OmegaConf.create(definition)
+    return definition
+
+
+# Parameters for tests of the "singleElement" rule when there is no interpolation.
```

Comment:
The rest of the tests from here seems like they belong in the grammar tests and not in this file.

Reply:
Well these are the only grammar tests I'm aware of. Do you mean I should move them into a new file? (ex: test_grammar.py)

Reply:
Yes. If there isn't already a file to test file for the grammar create one.

Reply:
Moved all these tests to a new file in e522aec